src/main.cpp

```diff
@@ -2636,22 +2645,28 @@ void txRxProcessing()
             int nread = codec2_fifo_read(cbData->infifo2, insound_card, nsam_in_48);
             if (nread != 0 && endingTx) break;
             
-            nout = resample(cbData->insrc2, infreedv, insound_card, freedvInterface.getTxSpeechSampleRate(), g_soundCard2SampleRate, 10*N48, nsam_in_48);
-
             // optionally use file for mic input signal
             if (g_playFileToMicIn && (g_sfPlayFile != NULL)) {
-                int n = sf_read_short(g_sfPlayFile, infreedv, nout);
-                //fprintf(stderr, "n: %d nout: %d\n", n, nout);
-                if (n != nout) {
+                unsigned int nsf = nsam_in_48*g_sfTxFs/g_soundCard2SampleRate;
+                short        insf[nsf];
+                                
+                int n = sf_read_short(g_sfPlayFile, insf, nsf);
+                nout = resample(cbData->insrctxsf, insound_card, insf, g_soundCard2SampleRate, g_sfTxFs, nsam_in_48, n);
+                
+                if (nout == 0) {
                     if (g_loopPlayFileToMicIn)
                         sf_seek(g_sfPlayFile, 0, SEEK_SET);
                     else {
-                        // call stop/start play menu item, should be thread safe
+                        printf("playFileFromRadio finished, issuing event!\n");
                         g_parent->CallAfter(&MainFrame::StopPlayFileToMicIn);
                     }
                 }
+                
+                codec2_fifo_write(cbData->infifo2, insound_card, nout);
             }
-
+            
+            nout = resample(cbData->insrc2, infreedv, insound_card, freedvInterface.getTxSpeechSampleRate(), g_soundCard2SampleRate, 10*N48, nsam_in_48);
+                 
```

Comment:
Guess this line could have stayed above the mic file processing (in the original position), and we could have resampled the mic file signal to `freedvInterface.getTxSpeechSampleRate()`.  But I don't think it makes much difference - this is fine :slightly_smiling_face: